datafusion/physical-expr-common/src/aggregate/mod.rs

```diff
@@ -49,7 +51,10 @@ pub fn create_aggregate_expr(
     ignore_nulls: bool,
     is_distinct: bool,
 ) -> Result<Arc<dyn AggregateExpr>> {
-    debug_assert_eq!(sort_exprs.len(), ordering_req.len());
+    // sort_exprs is used for udaf accumulator only, so it is not neccessary
+    // if the aggregate function does not care about accumulator
+    // debug_assert_eq!(sort_exprs.len(), ordering_req.len());
```

Comment:
To avoid adding tons of redundant sort expressions for window test, I remove the assertion here.

Reply:
I think, this invariant is pretty important. It helps to catch some bugs otherwise quite hard to locate. Hence, I think we should keep this check in place.

Reply:
By the way, can I sent commits to this PR?. I can re-introduce this check.

Reply:
> By the way, can I sent commits to this PR?. I can re-introduce this check.

Sure!

Reply:
I think this is the only blocker of this PR? I have no other commits to push.

Reply:
Sent a commit, to re-introduce this check.